sdk/azidentity/azidentity.go

```diff
@@ -73,6 +75,41 @@ var getPublicClient = func(clientID, tenantID string, co *azcore.ClientOptions)
 	)
 }
 
+// resolveAdditionallyAllowedTenants returns a copy of tenants, simplified when tenants contains a wildcard
+func resolveAdditionallyAllowedTenants(tenants []string) []string {
+	if len(tenants) == 0 {
+		return nil
+	}
+	for _, t := range tenants {
+		// a wildcard makes all other values redundant
+		if t == "*" {
+			return []string{"*"}
+		}
+	}
+	cp := make([]string, len(tenants))
+	copy(cp, tenants)
+	return cp
+}
+
+// resolveTenant returns the correct tenant for a token request given a credential's configuration
+func resolveTenant(defaultTenant, reqTenant string, allowedTenants []string) (string, error) {
+	if reqTenant == "" || reqTenant == defaultTenant {
+		return defaultTenant, nil
+	}
+	if defaultTenant == "adfs" {
+		return "", errors.New("the authority doesn't support tenants")
```

Comment:
Should this error message include the term "adfs" someplace?

Reply:
Sure, that could help someone.